packages/git/src/browser/git-quick-open-service.ts

```diff
@@ -38,9 +39,55 @@ export class GitQuickOpenService {
         @inject(Git) protected readonly git: Git,
         @inject(GitRepositoryProvider) protected readonly repositoryProvider: GitRepositoryProvider,
         @inject(QuickOpenService) protected readonly quickOpenService: QuickOpenService,
-        @inject(MessageService) protected readonly messageService: MessageService
+        @inject(MessageService) protected readonly messageService: MessageService,
+        @inject(WorkspaceService) protected readonly workspaceService: WorkspaceService,
     ) { }
 
+    async clone(url?: string, folder?: string, branch?: string): Promise<void> {
+        if (!folder) {
+            const roots = await this.workspaceService.roots;
+            folder = roots[0].uri;
+        }
+
+        const __folder = folder;
+        if (url) {
+            return this.git.clone(url, { localUri: this.buildDefaultProjectPath(folder, url), branch: branch }).then(() => { });
+        }
+
+        const __this = this;
+        const cloneRepoModel: QuickOpenModel = {
+            onType(lookFor: string, acceptor: (items: QuickOpenItem[]) => void): void {
+                const dynamicItems: QuickOpenItem[] = [];
+                const suffix = "Press 'Enter' to confirm or 'Escape' to cancel.";
+                if (lookFor === undefined || lookFor.length === 0) {
+                    dynamicItems.push(new SingleStringInputOpenItem(`Please provide a Git repository URI. ${suffix}`, () => { }, () => false));
```

Comment:
I would show `URL` instead of `URI` to the user. What do you think?
`Please provide a Git repository URI.` -> `Please provide a Git repository URL.`

Reply:
yes
I wasn't sure
- In one side, we often clone from github providing a URL
- But on another side, URI is more generic. We can also clone from a folder in the file system 
    for instance
    ```
    git clone /projects/existing-git-project /projects/a-new-project
    ```